USM.Test/AlgorithmTest.cs

```diff
@@ -0,0 +1,509 @@
+            {
+                Ticket = 251,
+                Course = CSE25
+            };
+
+            Meeting CSE26Lecture = new Meeting()
+            {
+                MeetingType = MeetingType.Lecture,
+                Days = (Days.Tuesday | Days.Thursday),
+                StartTime = 1530,
+                EndTime = 1700,
+                Code = "A00"
+            };
+
+            Meeting CSE26Discussion1 = new Meeting()
+            {
+                MeetingType = MeetingType.Discussion,
+                Days = Days.Wednesday,
+                StartTime = 1000,
+                EndTime = 1050,
+                Code = "A02"
+            };
+            Meeting CSE26Discussion2 = new Meeting()
+            {
+                MeetingType = MeetingType.Discussion,
+                Days = (Days.Tuesday | Days.Thursday),
+                StartTime = 1700,
+                EndTime = 1750,
+                Code = "B02"
+            };
+            Course CSE26 = new Course()
+            {
+                CourseAbbreviation = "CSE26",
+                CourseName = "Intro / Discrete Mathematics",
+                Units = 4,
+                Description = ""
+            };
+            Section CSE26Section1 = new Section()
+            {
+                Ticket = 261,
+                Course = CSE26
+            };
+            Section CSE26Section2 = new Section()
+            {
+                Ticket = 262,
+                Course = CSE26
+            };
+
+            Meeting CSE27Lecture = new Meeting()
+            {
+                MeetingType = MeetingType.Lecture,
+                Days = (Days.Monday | Days.Wednesday),
+                StartTime = 1530,
+                EndTime = 1700,
+                Code = "A00"
+            };
+
+            Meeting CSE27Discussion1 = new Meeting()
+            {
+                MeetingType = MeetingType.Discussion,
+                Days = Days.Wednesday,
+                StartTime = 1000,
+                EndTime = 1050,
+                Code = "A02"
+            };
+            Meeting CSE27Discussion2 = new Meeting()
+            {
+                MeetingType = MeetingType.Discussion,
+                Days = (Days.Tuesday | Days.Thursday),
+                StartTime = 1530,
+                EndTime = 1700,
+                Code = "B02"
+            };
+            Course CSE27 = new Course()
+            {
+                CourseAbbreviation = "CSE27",
+                CourseName = "Intro / Discrete Mathematics",
+                Units = 4,
+                Description = ""
+            };
+            Section CSE27Section1 = new Section()
+            {
+                Ticket = 271,
+                Course = CSE27
+            };
+            Section CSE27Section2 = new Section()
+            {
+                Ticket = 272,
+                Course = CSE27
+            };
+
+            // Adding different sections to the course object
```

Comment:
I would try to organize this in another way, as this expands I think we're going to spend more and more time verifying that our test classes are constructed correctly instead of whether or not our algorithm is sound.